https://github.com/facebook/hhvm/issues/4103

From 5ede5bb6d8b5bb8ba094d874771c64b0480e7b7a Mon Sep 17 00:00:00 2001
From: Sara Golemon <sgolemon@fb.com>
Date: Fri, 31 Oct 2014 16:07:05 -0700
Subject: [PATCH] Revert "Silence warning from CMake Policy CMP0026"

Summary: This reverts commit https://github.com/facebook/hhvm/commit/5b8c73ab2bfb31d29d52c7bdba28bc58ba1dd066

Reviewed By: @ptarjan

Differential Revision: D1650299

Signature: t1:1650299:1414713932:a1335e585ff36e5686b9b18f6513e5f4995094ba
---
 CMake/HPHPFunctions.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMake/HPHPFunctions.cmake b/CMake/HPHPFunctions.cmake
index e65de8e..957b2a3 100644
--- a/CMake/HPHPFunctions.cmake
+++ b/CMake/HPHPFunctions.cmake
@@ -153,19 +153,20 @@ endfunction(embed_all_systemlibs)
 # Custom install function that doesn't relink, instead it uses chrpath to change it, if
 # it's available, otherwise, it leaves the chrpath alone
 function(HHVM_INSTALL TARGET DEST)
+  get_target_property(LOC ${TARGET} LOCATION)
   get_target_property(TY ${TARGET} TYPE)
   if (FOUND_CHRPATH)
     get_target_property(RPATH ${TARGET} INSTALL_RPATH)
     if (NOT RPATH STREQUAL "RPATH-NOTFOUND")
       if (RPATH STREQUAL "")
-        install(CODE "execute_process(COMMAND \"${CHRPATH}\" \"-d\" \"$<TARGET_FILE>\" ERROR_QUIET)")
+        install(CODE "execute_process(COMMAND \"${CHRPATH}\" \"-d\" \"${LOC}\" ERROR_QUIET)")
       else()
-        install(CODE "execute_process(COMMAND \"${CHRPATH}\" \"-r\" \"${RPATH}\" \"$<TARGET_FILE>\" ERROR_QUIET)")
+        install(CODE "execute_process(COMMAND \"${CHRPATH}\" \"-r\" \"${RPATH}\" \"${LOC}\" ERROR_QUIET)")
       endif()
     endif()
   endif()
   string(TOUPPER ${DEST} DEST_UPPER)
-  install(CODE "FILE(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_${DEST_UPPER}DIR}\" TYPE ${TY} FILES \"$<TARGET_FILE>\")")
+  install(CODE "FILE(INSTALL DESTINATION \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_${DEST_UPPER}DIR}\" TYPE ${TY} FILES \"${LOC}\")")
 endfunction(HHVM_INSTALL)
 
 function(HHVM_PUBLIC_HEADERS TARGET)
